packages/integrations/mdx/src/index.ts

```diff
@@ -168,7 +173,12 @@ export default function mdx(partialMdxOptions: Partial<MdxOptions> = {}): AstroI
 											import.meta.hot.decline();
 										}`;
 									}
-									return escapeViteEnvReferences(code);
+									code = escapeViteEnvReferences(code);
+									if (config.vite.build?.sourcemap) {
+										const s = new MagicString(code);
+										return { code: s.toString(), map: s.generateMap() };
+									}
```

Comment:
This part does transform the code so it needs a map, but it's hard to generate a mapping here. Using `MagicString` isn't quite accurate too I believe. I think we could do `return { code, map: null }` to preserve the mapping as suggested in the linked rollup docs.

The transformation in `escapeViteEnvReferences` does column-based changes, so it should be fine to preserve tha sourcemap.

Reply:
Understood. Changed!